src/collection.rs

```diff
@@ -23,6 +26,143 @@ use strategy::*;
 use tuple::TupleValueTree;
 use test_runner::*;
 
+// TODO: add more CollectionStrategy impls for different types such as
+// HashSet.
+
+// TODO: tests for CollectionStrategy impls.
+
+/// Returns a `Strategy` that generates subsequences of `source` where the
+/// where the subsequences have a size: `size.start <= seq.len() < size.end`.
+pub fn subsequence<T>(source: Vec<T>, size: Range<usize>) -> SubseqStrategy<T>
+where
+    T: Clone + fmt::Debug
+{
+    source.prop_sample(size)
+}
+
+/// A collection, such as `Vec<T>` is itself a strategy to create samples
+/// (or in the case of `Vec`: subsequences) with the elements of the collection.
+pub trait CollectionStrategy: Sized + fmt::Debug {
+    /// The `Strategy` generating the samples.
+    type Strategy: Strategy<Value = Self::ValueTree>;
+
+    /// Always the same as `Self::Strategy::Value`.
+    /// Unfortunately this can't be removed until `impl Trait` lands.
+    type ValueTree: ValueTree<Value = Self>;
+
+    /// Constructs a `Strategy` to generate samples of `self` given a range
+    /// (low..high) where the output sample has a number of elements: len.
+    /// The following always holds for the sample: `low <= len < high`.
+    ///
+    /// # Safety
+    ///
+    /// A panic must occur at least when:
+    ///
+    /// 1. `low > high`,
+    /// 2. `self.len() < low + 1`.
+    fn prop_sample(self, size: Range<usize>) -> Self::Strategy;
+}
+
+impl<T: Clone + fmt::Debug> CollectionStrategy for Vec<T> {
+    type Strategy = SubseqStrategy<T>;
+    type ValueTree = SubseqValueTree<T>;
+    fn prop_sample(self, size: Range<usize>) -> Self::Strategy {
+        SubseqStrategy::new(self, size)
+    }
+}
+
+/// A `Strategy` to generate subsequences, of a `Vec<T>`,
+/// where the subsequences have a size: `min <= seq.len() < max`.
```

Comment:
again, could this be `min <= seq.len() <= max`

Reply:
See comment above.